core/src/main/java/org/bitcoinj/base/internal/ByteUtils.java

```diff
@@ -385,6 +398,17 @@ public static long readUint32(ByteBuffer buf) throws BufferUnderflowException {
         return Integer.toUnsignedLong(buf.order(ByteOrder.LITTLE_ENDIAN).getInt());
     }
 
+    /**
+     * Read a 4-byte integer from the buffer in little endian format.
+     * The integer may be signed or unsigned, the bytes are just read as is.
+     * @param buf buffer to be read from
+     * @return A 4-byte integer (that may be signed or unsigned depending upon the source.)
```

Comment:
For reading, there is this similar case we have above, just with the difference that we can't use an overload.

So again, the problem I have with the documentation is that it claims "the integer may be unsigned", but then it doesn't return a long for that case.